pkg/remotewrite/config.go

```diff
@@ -253,6 +286,8 @@ func GetConsolidatedConfig(jsonRawConf json.RawMessage, env map[string]string, a
 		}
 	}
 
+	result.Headers = getEnvMap(env, "K6_PROMETHEUS_HEADERS_")
```

Comment:
I think it's fine to overwrite values from previous config applications in case of headers but only as addition of values, not full overwrite of the map. So please change here:
```suggestion
	envHeaders := getEnvMap(env, "K6_PROMETHEUS_HEADERS_")
	for k, v := range envHeaders {
		result.Headers[k] = v
	}
```